### PR TITLE
Refs #306 ensuring user is active when trying to reset password

### PR DESCRIPTION
**Pull request by pedrofs**
It fixes the #306 

I've added one more option, `ensureActive`, to `PasswordBehavior#resetToken`.

Then `PasswordManagementTrait#requestResetPassword` passes the option when calling the `PasswordBehavior#resetToken`.

Sounds enough?

### CODE REVIEW

File: src/Model/Behavior/PasswordBehavior.php [hunk 1]
```diff
@@ -71,6 +72,11 @@ public function resetToken($reference, array $options = [])
             $user->active = false;
             $user->activation_date = null;
         }
+        if (Hash::Get($options, 'ensureActive')) {
```
Comment (steinkel) [inline src/Model/Behavior/PasswordBehavior.php]: should be Hash::get?

File: src/Model/Behavior/PasswordBehavior.php [hunk 2]
```diff
@@ -71,6 +72,11 @@ public function resetToken($reference, array $options = [])
...
+            if (!$user->active) {
```
Comment (steinkel) [inline src/Model/Behavior/PasswordBehavior.php]: array access is preferred

File: src/Model/Behavior/PasswordBehavior.php [hunk 3]
```diff
@@ -71,6 +72,11 @@ public function resetToken($reference, array $options = [])
...
+                throw new UserNotActiveException("User not active");
```
Comment (steinkel) [inline src/Model/Behavior/PasswordBehavior.php]: missing __d()

File: src/Model/Behavior/PasswordBehavior.php (hunk 2 above)
Comment (pedrofs) [inline src/Model/Behavior/PasswordBehavior.php]: hi @steinkel 

why using array access here?

**Comment by steinkel** [comment]
Please ensure coveralls is happy

**Comment by steinkel** [comment]
@pedrofs please check failing tests

**Comment by pedrofs** [comment]
@steinkel it is something related to postgresql, I'll investigate what is going on. btw, if you have any clue regarding this, thanks!

**Comment by steinkel** [comment]
Thank you!

**Pull request merged**